base/shell/explorer/notifyiconscust.cpp

```diff
@@ -20,31 +21,244 @@
+    lvItem.cchTextMax = 100;
+    lvItem.pszText = szBehavior;
+    ListView_SetItem(hListView, &lvItem);
+
+    ZeroMemory(&lvItem, sizeof(lvItem));
+    lvItem.mask = LVIF_PARAM;
+    lvItem.iSubItem = 0;
+    lvItem.lParam = iState;
+    lvItem.iItem = iItem;
+    ListView_SetItem(hListView, &lvItem);
+
+    ShowWindow(hCombo, SW_HIDE);
+}
+
+VOID ResetBehaviors(HWND hwnd)
+{
+    HWND hListView = GetDlgItem(hwnd, IDC_NOTIFICATION_LIST);
+    int count = ListView_GetItemCount(hListView);
+    WCHAR szBehavior[100];
+    LoadStringW(NULL, IDS_NOTIF_BEH_HIDE_INACTIVE, szBehavior, _countof(szBehavior));
+
+    LVITEM lvi;
+    ZeroMemory(&lvi, sizeof(lvi));
+    lvi.cchTextMax = _countof(szBehavior);
+    lvi.pszText = szBehavior;
+    lvi.lParam = 2;
+
+    for (int i = 0; i < count; i++)
+    {
+        lvi.mask = LVIF_PARAM;
+        lvi.iItem = i;
+        lvi.iSubItem = 0;
+        ListView_SetItem(hListView, &lvi);
+
+        lvi.mask = LVIF_TEXT;
+        lvi.iItem = i;
+        lvi.iSubItem = 1;
+        ListView_SetItem(hListView, &lvi);
+    }
+}
+
+VOID InitializeListView(HWND hwnd)
+{
+    HWND hListView = GetDlgItem(hwnd, IDC_NOTIFICATION_LIST);
+
+    LVCOLUMN lvColumn;
+    WCHAR sName[256];
+                
+    SendMessage(hListView, LVM_SETEXTENDEDLISTVIEWSTYLE, 0, LVS_EX_FULLROWSELECT);
+
+    ZeroMemory(&lvColumn, sizeof(lvColumn));
+    lvColumn.mask = LVCF_TEXT | LVCF_WIDTH | LVCF_SUBITEM;
+                
+    LoadStringW(NULL, IDS_NOTIF_BEH_NAME, sName, _countof(sName));
+    lvColumn.pszText = sName;
+    lvColumn.cx = 190;
+    ListView_InsertColumn(hListView, 0, &lvColumn);
+
+    LoadStringW(NULL, IDS_NOTIF_BEH_BEHAVIOR, sName, _countof(sName));
+    lvColumn.pszText = sName;
+    lvColumn.cx = 130;
+    ListView_InsertColumn(hListView, 1, &lvColumn);
+
+    ListView_EnableGroupView(hListView, TRUE);
+
+    LVGROUP lvGroup;
+    lvGroup.cbSize    = sizeof(LVGROUP);
+    lvGroup.mask      = LVGF_HEADER | LVGF_ALIGN | LVGF_GROUPID | LVGF_STATE;
+    lvGroup.uAlign    = LVGA_FOOTER_LEFT;
+    lvGroup.state     = LVGS_NORMAL;
+
+    LoadStringW(NULL, IDS_NOTIF_BEH_CURRENT, sName, _countof(sName));
+    lvGroup.pszHeader = sName;
+    lvGroup.iGroupId  = 0;
+    ListView_InsertGroup(hListView, -1, &lvGroup);
+
+    LoadStringW(NULL, IDS_NOTIF_BEH_PAST, sName, _countof(sName));
+    lvGroup.pszHeader = sName;
+    lvGroup.iGroupId  = 1;
+    ListView_InsertGroup(hListView, -1, &lvGroup);
+}
+
 INT_PTR CALLBACK CustomizeNotifyIconsProc(HWND hwnd, UINT Message, WPARAM wParam, LPARAM lParam)
 {
     switch(Message)
     {
-        case WM_INITDIALOG:
-        
-        return TRUE;
+        case WM_INITDIALOG: 
+            InitializeListView(hwnd);
+            break;
+        case WM_NOTIFY:
+            if (LOWORD(wParam) == IDC_NOTIFICATION_LIST && ((LPNMHDR)lParam)->code == NM_CLICK) 
+            {
+                ShowBehaviorCombo(hwnd, (LPNMITEMACTIVATE)lParam);
+            }
+            else return FALSE;
```

Comment:
Because first if-block was inside brackets, the return-FALSE should be inside brackets too (cf. our coding style).

Reply:
ok